src/components/OverlaysProvider/index.tsx

```diff
@@ -0,0 +1,25 @@
+/* eslint-disable @typescript-eslint/no-empty-function */
+import OverlayConfirm, { ConfirmProps, defaultConfirmProps } from './OverlayConfirm';
+import React, { useState } from 'react';
+
+interface OverlayContextProps {
+    setConfirmProps: (newProps: ConfirmProps) => void;
+}
+
+const defaultValue = {
+    setConfirmProps: () => null,
+};
+
+export const OverlayContext = React.createContext<OverlayContextProps>(defaultValue);
+
+export const OverlaysProvider: React.FunctionComponent = (props) => {
+    const [confirmProps, setConfirmProps] = useState(defaultConfirmProps);
+
+    return (
+        <OverlayContext.Provider
+            value={{ setConfirmProps }}>
+            <OverlayConfirm { ...confirmProps } />
+            { props.children }
+        </OverlayContext.Provider>
+    );
+};
```

Comment:
Let's put this in `src/hooks` and call it `DialogContext`, rather than `OverlayProvider`, reason being that dialog keeps the naming consistent with the MUI components we implement and the `ZetkinDialog`. Also, since the provider will be accessed just once in `_app.tsx`, the context is being accessed many places, it makes more sense to call the file `DialogContext` 

I would also do this in a folder with the `OverlayConfirmDialog` in there, and rename this file to `index.tsx`.